pubsub/streaming-analytics/PubSubToGCS.py

```diff
@@ -0,0 +1,105 @@
+import argparse
+import datetime
+import logging
+
+import apache_beam as beam
+import apache_beam.transforms.window as window
+from apache_beam.options.pipeline_options import PipelineOptions
+
+
+class ProcessDoFn(beam.DoFn):
+    """A distributed processing function that acts on individual elements in
+    the PCollection.
+    """
+
+    def process(self, element, publish_time=beam.DoFn.TimestampParam):
+        """Enrich each Pub/Sub message with its publish timestamps.
+
+        Args:
+            element (bytestring): Pub/Sub message.
+            publish_time: Default to the publish timestamp from Pub/Sub.
+
+        Yields:
+            str: Message body and publish timestamp.
+        """
+        publish_time = datetime.datetime.utcfromtimestamp(
+            float(publish_time)).strftime("%Y-%m-%d %H:%M:%S.%f")
+        yield element.decode('utf-8') + ' published at ' + publish_time
```

Comment:
How about making this into a dict? That would also make it necessary to format as string when writing to GCS which will most likely happen in any real world scenario. I think it's also pretty common to use dictionaries as an intermediate data structure, so it might be a good pattern to showcase.

```
yield {
  'message': element.decode('utf-8'),
  'publish_time': datetime.datetime.utcfromtimestamp(float(publish_time)),
}
```